api/src/Page/Shipment.php

```diff
@@ -2761,9 +2933,32 @@ function _create_awb()
 
         $ids = range(2, sizeof($this->arg('DEWARS')) + 1);
         $args = array_merge(array($ship['SHIPPINGID']), $this->arg('DEWARS'));
-        $dewars = $this->db->pq("SELECT d.dewarid, d.weight, IF(d.facilitycode, d.facilitycode, d.code) as name
-                FROM dewar d
-                WHERE d.shippingid=:1 AND d.dewarid IN (:" . implode(',:', $ids) . ")", $args);
+
+        // Update this query to get num pucks, num pins
+        $dewars = $this->db->pq(
+            "SELECT d.dewarid, d.weight, IF(d.facilitycode, d.facilitycode, d.code) as name, count(distinct c.containerId) as num_pucks, count(b.blsampleId) as num_samples
+            FROM dewar d
+            LEFT JOIN container c on c.dewarid = d.dewarid
+            LEFT JOIN BLSample b on b.containerId = c.containerId
+            WHERE d.shippingid=:1 AND d.dewarid IN (:" . implode(',:', $ids) . ")",
+            $args
+        );
+
+        if (
+            Utils::getValueOrDefault($use_shipping_service_incoming_shipments)
+            && in_array($this->arg('COUNTRY'), $facility_courier_countries)
+            && Utils::getValueOrDefault($use_shipping_service_redirect_incoming_shipments)
+        ) {
+            try {
+                $this->_create_shipment_shipment_request($ship, $dewars);
+                $this->_output(array('EXTERNAL' => "1"));
+                return;
+            } catch (\Exception $e) {
+                $error_response = json_decode($e->getMessage());
+                $this->_error($error_response->content->detail, $error_response->status);
+            }
+        }
+        $this->_error(json_encode($dewars));
```

Comment:
Should this error message contain more data? Right now it only contains a list of the dewars sent